Formula/curl.rb

```diff
@@ -22,24 +23,25 @@ class Curl < Formula
   keg_only :provided_by_macos
 
   depends_on "pkg-config" => :build
-
-  uses_from_macos "zlib"
-
-  on_linux do
-    depends_on "openssl@1.1"
-  end
+  depends_on "nghttp2"
+  depends_on "openssl@1.1"
 
   def install
     system "./buildconf" if build.head?
 
+    openssl = Formula["openssl@1.1"]
     args = %W[
       --disable-debug
       --disable-dependency-tracking
       --disable-silent-rules
       --prefix=#{prefix}
       --with-secure-transport
-      --without-ca-bundle
-      --without-ca-path
+      --with-ca-bundle=#{openssl.pkgetc}/cert.pem
+      --with-ca-path=#{openssl.pkgetc}/certs
+      --with-gssapi
+      --with-ssl=#{openssl.opt_prefix}
```

Comment:
Both `--with-ssl` and `--with-secure-transport`? How does that behave?

If we're going to use OpenSSL here though, it would be worth bringing `curl-openssl` into the discussion.

Reply:
Deleted `--with-secure-transport` since it appears to conflict with the `HTTPS-proxy` feature

Relating to `curl-openssl` (from #57795):

> > Maybe we should replace the current curl formula with curl-openssl
> 
> We won't, they serve very different goals. `curl` is supposed to be a newer replacement for `/usr/bin/curl` and `curl-openssl` is there mostly to coexist with other tools that require `curl` with additional features and/or openssl.

Reply:
`curl-openssl` was originally introduced because of the removal of options. It's a bit of a horrible hack IMO. The "additional features" part only came later as the existence of a separate formula was seen as an opportunity to enable them. I'm not saying it should definitely go, but it is nevertheless a discussion point I wanted to raise here.

It was needed for the likes of PHP because of mixed dependencies causing crashes, e.g. PHP using OpenSSL and Curl not doing so.

Reply:
I think the default curl should follow the formula rules about being in line with other vendors and upstream recommendation here and follow what apple does in terms of SSL lib for the least surprises